### PR TITLE
tests: logging: Add stress test

### PR DESCRIPTION
**Pull request by nordic-krch**
Add stress test for logging to check that it is coherent when logging from various contexts. Logging coherency means that each message ends up processed by the backend or adds up to dropped messages which are notified to backends.

Test is using ztress framework. Basic test is executed on all emulation platforms but real test (with higher sys clock frequency to generate more preemptions) is executed on selected platforms (some qemus are slow and may timeout).

Additionally, converted log_core to use spin lock.

### CODE REVIEW

**Comment by nordic-krch** [comment]
`DNM` until  #41969 is merged.

**Review by dcpleung — DISMISSED** [review]

File: tests/subsys/logging/log_stress/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+common:
+  filter: CONFIG_QEMU_TARGET and not CONFIG_SMP
+  tags: log_api logging
+  integration_platforms:
+    - qemu_x86
+tests:
+  logging.log_stress_light_v1:
+    extra_configs:
+      - CONFIG_LOG1=y
+      - CONFIG_LOG_MODE_DEFERRED=y
```
Comment (tejlmand) [inline tests/subsys/logging/log_stress/testcase.yaml:10]: minor nit. from what I can tell `CONFIG_LOG_MODE_DEFERRED=y` is used on all tests.

**Review by tejlmand — APPROVED** [review]
lgtm, a minor nit, but can stay in testcase.yaml if you prefer.

**Review by nashif — APPROVED** [review]

**Comment by ycsin** [comment]
merged but reverted? https://github.com/zephyrproject-rtos/zephyr/commit/49d08c25fe86b2849f331a28dd73714a7c1f1e88

**Pull request merged**